### PR TITLE
bpo-32096: Remove obj and mem from _PyRuntime

### PR DESCRIPTION
**Pull request by vstinner**
bpo-32096, bpo-30860:  Partially revert the commit
2ebc5ce42a8a9e047e790aefbf9a94811569b2b6:

* Move structures back from Include/internal/mem.h to
  Objects/obmalloc.c
* Remove _PyObject_Initialize() and _PyMem_Initialize()
* Remove Include/internal/pymalloc.h

PyMem_RawMalloc() and Py_DecodeLocale() can be called again before
_PyRuntimeState_Init().

https://bugs.python.org/issue32096

### CODE REVIEW

**Review by ncoghlan — DISMISSED** [review]
While I'm +1 for this reversion, it should have a NEWS entry, since it's reverting changes that were published in previous alpha releases.

It should also incorporate @ericsnowcurrently's new test case from https://github.com/python/cpython/pull/4495/files#diff-559231a69a7d7f06ab48269b3bfa1f45, to confirm that it actually is fixing the regression it's intended to fix.

File: Misc/NEWS.d/next/Core and Builtins/2017-11-24-01-13-58.bpo-32096.CQTHXJ.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Revert memory allocator changes in the C API: move structures back from
+_PyRuntime to Objects/obmalloc.c. The memory allocators are now initialized
+again statically, and so PyMem_RawMalloc() and Py_DecodeLocale() can be
```
Comment (ncoghlan) [inline Misc/NEWS.d/next/Core and Builtins/2017-11-24-01-13-58.bpo-32096.CQTHXJ.rst]: Slight grammar tweak: "now initialized again" -> "once again initialized"

**Review by ncoghlan — APPROVED** [review]
LGTM, thanks!

File: Misc/NEWS.d/next/Core and Builtins/2017-11-24-01-13-58.bpo-32096.CQTHXJ.rst (hunk 1 above)
Comment (vstinner) [reply Misc/NEWS.d/next/Core and Builtins/2017-11-24-01-13-58.bpo-32096.CQTHXJ.rst]: Fixed

**Pull request merged**